authorizerd.go

```diff
@@ -139,24 +150,50 @@ func (p *provider) StartProviderd(ctx context.Context) <-chan error {
 }
 
 // VerifyRoleToken verifies the role token for specific resource and return and verification error.
-func (p *provider) VerifyRoleToken(ctx context.Context, tok, act, res string) error {
+func (p *authorizer) VerifyRoleToken(ctx context.Context, tok, act, res string) error {
+	return p.verify(ctx, token, tok, act, res)
+}
+
+func (p *authorizer) VerifyRoleJWT(ctx context.Context, tok, act, res string) error {
+	return p.verify(ctx, jwt, tok, act, res)
+}
+
+func (p *authorizer) verify(ctx context.Context, m mode, tok, act, res string) error {
```

Comment:
For readability, suggest not to use `mode` variable to determine which type of credential is verifying, instead separate the logic of checking the cache to new function.
Also variable `tok` should be `cred` or other meaningful name.
```suggestion
func (p *authorizer) verify(ctx context.Context, cred, act, res string) error {
```

Reply:
mode is not Public user don't need think about it

Reply:
How to decide logic ? when remove mode argument

Reply:
However the user needs to think about what credential to be verify, in my suggestion it decide by the function user called... My concern is that whenever the type of verifiable is added, the switch statement will required to change.

```golang
// VerifyRoleToken verifies the role token for specific resource and return and verification error.
func (p *authorizer) VerifyRoleToken(ctx context.Context, tok, act, res string) error {
       if p.successBefore(ctx, .....) {
             return nil
       }

      // verify token logic
     // if success then put it in success cache
}


func (p *authorizer) VerifyRoleJWT(ctx context.Context, tok, act, res string) error {
	// about the same as verify role token logic.
}

func (p *authorizer) successBefore(..... ) bool {
         // check if exists in verification success cache
	_, ok := p.cache.Get(tok + act + res)
	if ok {
		glg.Debugf("use cached result. tok: %s, act: %s, res: %s", tok, act, res)
	}
       return ok
}
```

Reply:
Yes user should think about what credential to be verify. Because they decide which function should be use for there workload.
verify is private function user don't need think about it.

I understand your concern, but I think it is more easy to extend switch statement than add new logic using minimum function assets.